### ISSUE TITLE
plugin doesn't detect a non zero exit status returned from minion cmd.run function

### ISSUE THREAD

**Issue by zerthimon** [jenkinsci/saltstack-plugin#18]
Salt detects this well:

```
root@master:~# salt minion cmd.run 'true 1'
minion:
root@master:~# echo $?
0
root@master:~# salt minion cmd.run 'false 1'
minion:
ERROR: Minions returned with non-zero exit code
root@master:~# echo $?
11
root@master:~#
```

The plugin doesn't detect the error and continues the job.
It would rock if plugin detects this situation and fails the job (or not ? config parameter ?)

**Comment by mchugh19** [comment]
Thanks for reporting. I'll get on it

**Comment by mchugh19** [comment]
Hmmm. It appears this isn't exposed through the salt-api. 

Here's the output from ```cmd.run 'false 1'```:
```
{
    "return": [
        {
            "web1": ""
        }
    ]
}
```

So first we'll need salt-api updated as mentioned in saltstack/salt/issues/18510

**Comment by zerthimon** [comment]
So, salt CLI command receives the exit status from minion and exits correctly, but salt-api doesn't expose it to the caller, right?

**Comment by mchugh19** [comment]
Yep. It looks that way. At least for 2015.8.8

**Comment by mchugh19** [comment]
The salt-api has been enhanced with the 2017.7 release, and this will be supported in the next release of the Jenkins plugin.

**Comment by zerthimon** [comment]
Hell yeah !!!

**Comment by mchugh19** [comment]
This was corrected in the 3.2.0 release. It should be available within the next 24 hours.
